src/lib/store/storeUtils.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useRef, useState } from 'react';
+import BaseStore from '@lib/store/config/baseStore';
+
+const storeUtils = <State, Action>(store: BaseStore<State, Action>) => {
+  const stateRef = useRef(store.getState());
+  const [, forceRender] = useState({});
+
+  const selector = <SelectedState>(selector: (state: State) => SelectedState): SelectedState => selector(stateRef.current);
+
+  const dispatch = (action: Action) => {
+    store.dispatch(action);
+    stateRef.current = store.getState();
```

Comment:
P2: 이쪽 forceRender 관련해서 고민이 있으셨을 것 같은데요, 이전 상태와 현재 상태를 확인하고 불필요한 랜더링을 개선하면 좋을 것 같아요

```suggestion
  const nextState = store.getState();
  if (stateRef.current !== nextState) {
    stateRef.current = nextState;
    forceRender({});
  }
```

Reply:
> P2: 이쪽 forceRender 관련해서 고민이 있으셨을 것 같은데요, 이전 상태와 현재 상태를 확인하고 불필요한 랜더링을 개선하면 좋을 것 같아요

이렇게 비교해서도 렌더링 처리할수 있겠네요!
피드백 감사합니다. 😄